sdlib/d/gc/tcache.d

```diff
@@ -320,6 +225,92 @@ public:
 
 private:
 
+	/**
+	 * Appendable's mechanics:
+	 * 
+	 *  __data__  _____free space_______
+	 * /        \/                      \
+	 * -----sss s....... ....... ........
+	 *      \___________________________/
+	 * 	           Capacity is 27
+	 * 
+	 * If the slice's end doesn't match the used capacity,
+	 * then we return 0 in order to force a reallocation
+	 * when appending:
+	 * 
+	 *  ___data____  ____free space_____
+	 * /           \/                   \
+	 * -----sss s---.... ....... ........
+	 *      \___________________________/
+	 * 	           Capacity is 0
+	 * 
+	 * See also: https://dlang.org/spec/arrays.html#capacity-reserve
+	 */
+
+	bool getOrExtendCapacity(const void[] slice, size_t size,
+	                         ref size_t sliceCapacity) {
+		auto pd = maybeGetPageDescriptor(slice.ptr);
+		auto e = pd.extent;
+		if (e is null) {
+			sliceCapacity = 0;
+			return false;
+		}
+
+		size_t usedCapacity;
+
+		size_t getSliceCapacity(const void[] space) {
```

Comment:
Don't use lambda in the GC code (outside of tests and alike). lambda need the GC to work. This is self defeating.